src/tests/test_dilithium.cpp

```diff
@@ -0,0 +1,295 @@
+							DRBG_ctx.V[j] = 0x00;
+						else {
+							DRBG_ctx.V[j]++;
+							break;
+						}
+					}
+					AES256_ECB(DRBG_ctx.Key, DRBG_ctx.V, block);
+					if (xlen > 15) {
+						memcpy(x + i, block, 16);
+						i += 16;
+						xlen -= 16;
+					}
+					else {
+						memcpy(x + i, block, xlen);
+						xlen = 0;
+					}
+				}
+				AES256_CTR_DRBG_Update(NULL, DRBG_ctx.Key, DRBG_ctx.V);
+				DRBG_ctx.reseed_counter++;
+
+				return 0;
+			}
+
+			typedef struct {
+				unsigned char   Key[32];
+				unsigned char   V[16];
+				int             reseed_counter;
+			} AES256_CTR_DRBG_struct;
+			AES256_CTR_DRBG_struct  DRBG_ctx;
+		};
+
+		Test::Result run_dilithium_kat_test(const char* test_name, const VarMap& vars, Botan::DilithiumDimension dimension, Botan::DilithiumAlgo algo,
+			[[maybe_unused]] const std::string& algo_name)
+		{
+			Test::Result result(test_name);
+			int                 ret_val;
+
+			// read input from test file
+			const auto ref_seed = vars.get_req_bin("seed");
+			const auto ref_mlen = vars.get_req_sz("mlen");
+			const auto ref_msg = vars.get_req_bin("msg");
+			const auto ref_pk = vars.get_req_bin("pk");
+			const auto ref_sk = vars.get_req_bin("sk");
+			auto ref_smlen = vars.get_req_sz("smlen");
+			const auto ref_sm = vars.get_req_bin("sm");
+			
+			// Kyber test RNG
+			std::unique_ptr<Botan::RandomNumberGenerator>dilithium_test_rng;
+			dilithium_test_rng.reset(new Dilithium_Test_RNG(ref_seed));
+
+			Botan::Dilithium_PrivateKey privKey(*dilithium_test_rng, dimension, algo);
+			result.test_eq("Private key generation Botan style equal with reference: ",
+				privKey.private_key_bits(), ref_sk);
+			result.test_eq("Public key generation Botan style equal with reference: ",
+				privKey.public_key_bits(), ref_pk);
+
+			auto signer = privKey.create_signature_op(*dilithium_test_rng, "raw", "");
+			signer->update(ref_msg.data(), ref_mlen);
+			auto signature = signer->sign(*dilithium_test_rng);
+			std::vector<uint8_t> ref_signature(ref_sm.begin(), ref_sm.begin() + (ref_smlen - ref_mlen));
+			result.test_eq("Signed Message Botan style equal with reference: ", signature,
+				ref_signature);
+			result.test_eq("Signed Message Length botan style equal with reference: ",
+				signature.size(), ref_signature.size());
+
+			Botan::Dilithium_PublicKey pub_key(privKey.public_key_bits(), dimension, algo, Botan::DilithiumKeyEncoding::Raw);
+			auto verificator = pub_key.create_verification_op("", "");
+			verificator->update(ref_msg.data(), ref_mlen);
+			ret_val = verificator->is_valid_signature(signature.data(), signature.size());
+			result.test_is_eq("Signature Verification: ", ret_val, 1);
+
+			// wrong signagture
+			signature[0] = 0;
+			signature[1] = 0; 
+			signature[2] = 0;
+			ret_val = verificator->is_valid_signature(signature.data(), signature.size());
+			result.test_is_eq("Wrong Signature Verification: ", ret_val, 0);
+
+			std::tuple<std::string, std::string> standardized_name_oid = [dimension, algo]
```

Comment:
This test doesn't need to be re-run for every single KAT. Could just be a one-off test.